components/kyma-environment-broker/internal/edp/client_builder.go

```diff
@@ -0,0 +1,40 @@
+package edp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+const (
+	namespaceToken = "%s/oauth2/token"
+)
+
+type Config struct {
+	AuthURL     string
+	AdminURL    string
+	Namespace   string
+	Secret      string
+	Environment string `envconfig:"default=prod"`
+	Required    bool   `envconfig:"default=false"`
+	Disabled    bool
+}
+
+func CreateEDPAdminClient(config Config, log logrus.FieldLogger) *Client {
```

Comment:
I tried to avoid creating oAuth client inside EDP client constructor, thanks DI I can simply in test inject client from `httptest.NewServer`, I couldn't find oAuth client test. If you see any simple solution about how to create oAuth inside EDP client and easily handle it in the test not implement all oAuth paths then I am open to suggestions.

Reply:
let's discuss that f2f. I know some trick for such tests :)